packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -611,12 +611,29 @@ function dbCollectionGroupKey(
     /* document id */ path.length > 0 ? path[path.length - 1] : ''
   ];
 }
+
 /**
  * Comparator that compares document keys according to the primary key sorting
- * used by the `DbRemoteDocumentDocument` store (by collection path and then
- * document ID).
+ * used by the `DbRemoteDocumentDocument` store (by prefix path, collection id
+ * and then document ID).
+ *
+ * Visible for testing.
```

Comment:
Have you been able to find a spec on IDB ordering? If yes, we can add it here.

Reply:
I have not and my contact has left the IndexedDb team. Let me spend some more time on research here though.

Reply:
I found https://chromium.googlesource.com/chromium/blink/+/refs/heads/main/Source/modules/indexeddb/IDBKey.cpp#66

So we are doing the right thing. I will add the link once we decide on the second comment thread in the PR.